### DISCUSSION TITLE
Port xorg-xinput

### DISCUSSION THREAD

**Discussion by jabcross** [termux/termux-packages#19426] [Ideas]
termux-x11 is *almost* perfect with my setup (Samsung Galaxy Tab S7 FE), but the touchpad coordinates are rotated 90 degrees. Looking for a solution online, I found that a generic fix uses xinput to add a coordinate transform matrix to the device. However, xorg-xinput is not available in the termux repository. I'm having trouble finding another fix (changing things in xorg.conf.d doesn't help).

How difficult would it be to port xinput in a way that it's compatible with termux-x11?

**Comment by Jacobw1oo** [comment]
Would this translate you to the subsystems? like to a rootfs and wine.

**Reply by jabcross** [reply]
Not sure what you mean? I thought xinput was just an interface to talk to xorg APIs, which I expect Termux:X11 already implements.

**Comment by Jacobw1oo** [comment]
I tried to build it my self but it needed xorg-marcos. i could not configure it with out rooting android.

**Reply by jabcross** [reply]
My device is rooted, so this is not a problem for me. Did you get them from the official repos?

**Reply by Jacobw1oo** [reply]
xorg-marcos is missing and i cant compile it, do to so many other thing missing.

**Comment by knyipab** [comment]
Seems relate to https://github.com/termux/termux-x11/issues/419
Probably can be fixed by https://github.com/termux/termux-x11/pull/617
And action apk build at https://github.com/termux/termux-x11/actions/runs/9222692955
